libraries/lib-update-version/UpdateManager.cpp

```diff
@@ -0,0 +1,217 @@
+#include "update/UpdateManager.h"
+#include "update/UpdatePopupDialog.h"
+
+#include <fstream>
+#include <iostream>
+
+#include "xml/XMLFileReader.h"
+#include "widgets/AudacityMessageBox.h"
+
+#include <wx/platinfo.h>
+#include <wx/utils.h>
+
+static const char* prefsUpdatePopupDialogShown = "/Update/UpdatePopupDialogShown";
+
+enum { ID_TIMER = wxID_HIGHEST + 1 };
+
+BEGIN_EVENT_TABLE(UpdateManager, wxEvtHandler)
+    EVT_TIMER(ID_TIMER, UpdateManager::OnTimer)
+END_EVENT_TABLE()
+
+UpdateManager::UpdateManager (AudacityProject& project)
```

Comment:
UpdateManager doesn't really use Project except to find its parent window.

Better then, that you simply construct it with a window.  (Passed by reference not pointer if null is not permittted.)